File: claude/.claude/skills/shortcut/scripts/objectives.py
#!/usr/bin/env python3
"""Shortcut Objectives - Operations for objective management."""

import sys
import json
import argparse
from typing import Dict, List, Optional
from shortcut_client import ShortcutClient, format_objective


def get_objective(objective_id: str) -> Dict:
    """Get a single objective by ID."""
    client = ShortcutClient()
    objective = client.get(f"objectives/{objective_id}")
    return format_objective(objective)


def list_objectives() -> List[Dict]:
    """List all objectives."""
    client = ShortcutClient()
    objectives = client.get("objectives")
    return [format_objective(obj) for obj in objectives]


def create_objective(
    name: str,
    description: Optional[str] = None
) -> Dict:
    """Create a new objective."""
    client = ShortcutClient()

    objective_data = {"name": name}
    if description:
        objective_data["description"] = description

    objective = client.post("objectives", objective_data)
    return format_objective(objective)


def update_objective(
    objective_id: str,
    name: Optional[str] = None,
    description: Optional[str] = None,
    state: Optional[str] = None
) -> Dict:
    """Update an existing objective."""
    client = ShortcutClient()

    update_data = {}
    if name is not None:
        update_data["name"] = name
    if description is not None:
        update_data["description"] = description
    if state is not None:
        update_data["state"] = state

    objective = client.put(f"objectives/{objective_id}", update_data)
    return format_objective(objective)


def delete_objective(objective_id: str) -> Dict:
    """Delete an objective."""
    client = ShortcutClient()
    client.delete(f"objectives/{objective_id}")
    return {"success": True, "message": f"Objective {objective_id} deleted"}


def main():
    parser = argparse.ArgumentParser(description="Shortcut Objectives Operations")
    subparsers = parser.add_subparsers(dest="command")

    get_parser = subparsers.add_parser("get")
    get_parser.add_argument("objective_id")

    subparsers.add_parser("list")

    create_parser = subparsers.add_parser("create")
    create_parser.add_argument("name")
    create_parser.add_argument("--description")

    update_parser = subparsers.add_parser("update")
    update_parser.add_argument("objective_id")
    update_parser.add_argument("--name")
    update_parser.add_argument("--description")
    update_parser.add_argument("--state")

    delete_parser = subparsers.add_parser("delete")
    delete_parser.add_argument("objective_id")

    args = parser.parse_args()

    try:
        if args.command == "get":
            result = get_objective(args.objective_id)
        elif args.command == "list":
            result = list_objectives()
        elif args.command == "create":
            result = create_objective(args.name, args.description)
        elif args.command == "update":
            result = update_objective(
                args.objective_id,
                args.name if hasattr(args, 'name') else None,
                args.description if hasattr(args, 'description') else None,
                args.state if hasattr(args, 'state') else None
            )
        elif args.command == "delete":
            result = delete_objective(args.objective_id)
        else:
            parser.print_help()
            sys.exit(1)

        print(json.dumps(result, indent=2))
    except Exception as e:
        print(json.dumps({"error": str(e)}, indent=2), file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()

File: claude/.claude/skills/shortcut/src/entities/objectives.ts
import type { Command } from "commander";
import type {
  ApiObjective,
  FormattedObjective,
  DeleteResult,
} from "../types.js";
import {
  ShortcutClient,
  formatObjective,
  output,
  outputError,
} from "../client.js";

export async function getObjective(
  objectiveId: string,
): Promise<FormattedObjective> {
  const client = new ShortcutClient();
  const objective = await client.get<ApiObjective>(`objectives/${objectiveId}`);
  return formatObjective(objective);
}

export async function listObjectives(): Promise<FormattedObjective[]> {
  const client = new ShortcutClient();
  const objectives = await client.get<ApiObjective[]>("objectives");
  return objectives.map(formatObjective);
}

export async function createObjective(
  name: string,
  description?: string,
): Promise<FormattedObjective> {
  const client = new ShortcutClient();

  const objectiveData: Record<string, unknown> = { name };
  if (description) objectiveData.description = description;

  const objective = await client.post<ApiObjective>(
    "objectives",
    objectiveData,
  );
  return formatObjective(objective);
}

export interface UpdateObjectiveOptions {
  objectiveId: string;
  name?: string;
  description?: string;
  state?: string;
}

export async function updateObjective(
  options: UpdateObjectiveOptions,
): Promise<FormattedObjective> {
  const client = new ShortcutClient();
  const { objectiveId, name, description, state } = options;

  const updateData: Record<string, unknown> = {};
  if (name !== undefined) updateData.name = name;
  if (description !== undefined) updateData.description = description;
  if (state !== undefined) updateData.state = state;

  const objective = await client.put<ApiObjective>(
    `objectives/${objectiveId}`,
    updateData,
  );
  return formatObjective(objective);
}

export async function deleteObjective(
  objectiveId: string,
): Promise<DeleteResult> {
  const client = new ShortcutClient();
  await client.delete(`objectives/${objectiveId}`);
  return { success: true, message: `Objective ${objectiveId} deleted` };
}

export function registerReadCommands(program: Command): void {
  const objectives = program
    .command("objectives")
    .description("Objective operations");

  objectives
    .command("get <objective-id>")
    .description("Get an objective by ID")
    .action(async (objectiveId: string) => {
      try {
        const result = await getObjective(objectiveId);
        output(result);
      } catch (e) {
        outputError(e);
      }
    });

  objectives
    .command("list")
    .description("List all objectives")
    .action(async () => {
      try {
        const result = await listObjectives();
        output(result);
      } catch (e) {
        outputError(e);
      }
    });
}

export function registerWriteCommands(program: Command): void {
  const objectives = program
    .command("objectives")
    .description("Objective operations");

  objectives
    .command("create <name>")
    .description("Create a new objective")
    .option("--description <desc>", "Objective description")
    .action(async (name: string, opts) => {
      try {
        const result = await createObjective(name, opts.description);
        output(result);
      } catch (e) {
        outputError(e);
      }
    });

  objectives
    .command("update <objective-id>")
    .description("Update an objective")
    .option("--name <name>", "Updated name")
    .option("--description <desc>", "Updated description")
    .option("--state <state>", "Updated state")
    .action(async (objectiveId: string, opts) => {
      try {
        const result = await updateObjective({
          objectiveId,
          name: opts.name,
          description: opts.description,
          state: opts.state,
        });
        output(result);
      } catch (e) {
        outputError(e);
      }
    });

  objectives
    .command("delete <objective-id>")
    .description("Delete an objective")
    .action(async (objectiveId: string) => {
      try {
        const result = await deleteObjective(objectiveId);
        output(result);
      } catch (e) {
        outputError(e);
      }
    });
}
